pkg/playback/http_playback.go

```diff
@@ -95,6 +95,8 @@ type Server struct {
 
 	log *log.Logger
 
+	SwitchModeChan chan string
```

Comment:
What do you think about  making the channel private and implementing a public `OnSwitchMode` method for Server that calls its argument? I try to encapsulate channels as much as possible because there's so much you can do with them (close them, listen for messages, send messages, wait for them to be closed) but usually you only intend for them to be used in a particular way (just listening for messages, in this case).

Reply:
That's a great point, I'll do that indeed!

Reply:
I'm not sure what is the best pattern to do that in Go, can you see what I have and let me know what you think? It doesn't feel very right to me but can't find a better way to handle that...